mongo/collection.go

```diff
@@ -1671,11 +1677,12 @@ func (coll *Collection) FindOne(ctx context.Context, filter interface{},
 			ShowRecordID:        opt.ShowRecordID,
 			Skip:                opt.Skip,
 			Sort:                opt.Sort,
-		})
+		}
 	}
+
 	// Unconditionally send a limit to make sure only one document is returned and the cursor is not kept open
 	// by the server.
-	findOpts = append(findOpts, options.Find().SetLimit(-1))
```

Comment:
Optional: The explicit indexing is somewhat confusing at first and seems brittle. Is there a reason to do that vs using `append`?

Reply:
I can't think of a valid reason not to do it with the append solution. I will update the code, thanks.

Reply:
The append method causes the following linting error:

```
mongo/collection.go:1667:14: append to slice `findOpts` with non-zero initialized length (makezero)
                findOpts = append(findOpts, &options.FindOptions{
                           ^
mongo/collection.go:1685:13: append to slice `findOpts` with non-zero initialized length (makezero)
        findOpts = append(findOpts, options.Find().SetLimit(-1))
```

Reply:
@matthewdale Looking at this more closely, I believe that the original solution and this solution both have a pretty insidious bug. Namely if a user passes options like this to the FindOne function: [op1, nil, op2], you would get an incorrect merge. In the solution I am proposing you'd actually get a runtime fatal. I've updated the PR with a test for this.

Reply:
Nice find!